src/io/flutter/run/bazel/FlutterBazelRunConfigurationType.java

```diff
@@ -32,14 +32,25 @@ public Factory(FlutterBazelRunConfigurationType type) {
     @Override
     @NotNull
     public RunConfiguration createTemplateConfiguration(@NotNull Project project) {
+      // This is always called first when loading a run config, even when it's a non-template config.
+      // See RunManagerImpl.doCreateConfiguration
       return new BazelRunConfig(project, this, "Flutter (Bazel)");
     }
 
     @Override
     @NotNull
     public RunConfiguration createConfiguration(String name, RunConfiguration template) {
-      // Override the default name which is always "Unnamed".
-      return super.createConfiguration(template.getProject().getName(), template);
+      // Called in two cases:
+      //   - When creating a non-template config from a template.
+      //   - whenever the run configuration editor is open (for creating snapshots).
+      // In the first case, we want to override the defaults from the template.
+      // In the second case, don't change anything.
+      //   (But the name doesn't matter; it will immediately be overwritten.)
+      if ((name.equals("Unnamed") || name.startsWith("Unnamed (")) && template instanceof BazelRunConfig) {
```

Comment:
I know you didn't add this here but I wonder if there isn't a less hacky way to test for the "Unnamed" case. (And is that name localized?)  Maybe @alexander-doroshko knows?